tests/unit/test_svs_tiered.cpp

```diff
@@ -386,6 +386,102 @@ TYPED_TEST(SVSTieredIndexTest, insertJobAsync) {
     }
 }
 
+/**
+ * This test verifies that a tiered index correctly returns the closest vectors when querying data
+ * distributed across both the flat and SVS indices, specifically when duplicate labels exist in
+ * both indices with different distances. It adds vectors with known scores, including such
+ * duplicates, and ensures that only the closer instance is returned. The test covers both top-K and
+ * range queries, validating result ordering by score and by ID.
+ */
+TYPED_TEST(SVSTieredIndexTest, SearchDifferentScores) {
+    size_t dim = 4;
+    size_t constexpr k = 3;
+
+    // Create TieredSVS index instance with a mock queue.
+    SVSParams params = {
+        .type = TypeParam::get_index_type(),
+        .dim = dim,
+        .metric = VecSimMetric_L2,
+    };
+    VecSimParams svs_params = CreateParams(params);
+    auto mock_thread_pool = tieredIndexMock();
+    auto *tiered_index = this->CreateTieredSVSIndex(svs_params, mock_thread_pool, k);
+    ASSERT_INDEX(tiered_index);
+
+    auto svs_index = tiered_index->GetBackendIndex();
+    auto flat_index = tiered_index->GetFlatIndex();
+
+    // Define IDs and distance values for test vectors
+    // ids are intentionally in random order to verify sorting works correctly
+    size_t constexpr ids[k] = {54, 4, 15};
+    double constexpr res_values[k] = {2, 3, 100};
+    // Define a type for our result pair
+    using ResultPair = std::pair<size_t, double>; // (id, score)
+
+    // Create a vector of expected results - these are the scores we expect
+    // when querying with a zero vector (L2 distance = value²*dim)
+    std::vector<ResultPair> expected_results_by_score(k);
+
+    for (size_t i = 0; i < k; i++) {
+        expected_results_by_score[i] = {ids[i], res_values[i] * res_values[i] * dim};
+    }
+
+    // Insert duplicate vectors with same ID but different distances across the two indices.
+    // The index should return only the closer of the two.
+
+    // ID 54: closer in SVS, farther in flat — expect to return SVS version
+    GenerateAndAddVector<TEST_DATA_T>(svs_index, dim, ids[0], res_values[0]);
+    GenerateAndAddVector<TEST_DATA_T>(flat_index, dim, ids[0], 4);
```

Comment:
As I understand, flat index should be prioritized in this case:
* user adds a vector with a value _version_1_ (moved to backend)
* user overrides the vector with a value _version_2_ (not yet moved to backend but kept in flat)
* user expects that _version_1_ is overridden and forgotten and will never appear in query results